app/models/link.rb

```diff
@@ -1,4 +1,11 @@
 class Link < ApplicationRecord
-  has_and_belongs_to_many :jobs
+  has_many :checks
+
   validates_presence_of :uri
+
+  def find_completed_check(within: 24.hours)
+    checks
+      .where("completed_at > ?", Time.now - within)
+      .first
+  end
```

Comment:
I think, but not sure, that this might be better as a scope on `Check` it'd help differentiate it as it executes a query

Should also have an ordering to get most recent one.